### PR TITLE
[DGB-627]  Implement the ability to create data tables for a transaction entity

### PR DESCRIPTION
**Pull request by y1zozulia**
Implemented for m_savings_account_transactional

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java [hunk 1]
```diff
@@ -1350,7 +1350,12 @@ private String dataScopedSQL(final String appTable, final Long appTableId) {
             scopedSQL = "select null as officeId, null as groupId, null as clientId, null as savingsId, null as loanId, p.id as entityId from "
                     + appTable + " as p WHERE p.id = " + appTableId;
         }
-
+        if (appTable.equalsIgnoreCase("m_savings_account_transaction")) {
+            scopedSQL = "select o.id as officeId, null as groupId, null as clientId, x.id as savingsId, null as loanId, null as entityId from "
+                    + appTable + " as t " + " join m_office o on o.id = t.office_id and o.hierarchy like '"
```
Comment (ysanko-tranzzo) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: Could we move `t.office_id and o.hierarchy like ...` to WHERE clause?

**Review by ysanko-tranzzo — APPROVED** [review]

**Review by ysanko-tranzzo — APPROVED** [review]

**Comment by awasum** [comment]
Hi @y1zozulia  and @ysanko-tranzzo . Welcome to the Fineract upstream project. Your PR looks like it solves a real problem but I will like to ask you to create a JIRA ticket on the open source issue system here: https://issues.apache.org/jira/projects/FINERACT/issues/, this way we have a public description of the issue you are solving and can review this PR with more context. Also edit your PR to include the FINERACT-xxx issue name as oppose to your internal issue tracking number: DGB-627. After that, we will be able to review this PR and get it merged if everything looks Ok. Does this make sense to you all?

**Comment by vorburger** [comment]
@ysanko-tranzzo I agree with @awasum that it would be nice to have a JIRA issue for this on the Apache JIRA, instead of _DGB-627_ which we're assuming is some internal issue tracking you use. Would you be willing to raise one, with a suitable description? Do you need any assistance to get started on the ASF JIRA?

**Comment by y1zozulia** [comment]
Hi @vorburger and @awasum. This PR need just only for our fork.

**Pull request closed without merging**